pkg/noderesourcetopology/pluginhelpers.go

```diff
@@ -80,30 +85,91 @@ func initNodeTopologyInformer(tcfg *apiconfig.NodeResourceTopologyMatchArgs, han
 }
 
 func createNUMANodeList(zones topologyv1alpha2.ZoneList) NUMANodeList {
-	nodes := make(NUMANodeList, 0, len(zones))
-	for _, zone := range zones {
+	// maxNUMAId is 64 just init array with this len
+	numaIDToZoneIDx := make([]int, maxNUMAId)
+	nodes := NUMANodeList{}
+	// filter non Node zones and create idToIdx lookup array
+	for i, zone := range zones {
 		if zone.Type != "Node" {
 			continue
 		}
-		var numaID int
-		_, err := fmt.Sscanf(zone.Name, "node-%d", &numaID)
+
+		numaID, err := getID(zone.Name)
 		if err != nil {
-			klog.ErrorS(nil, "Invalid zone format", "zone", zone.Name)
-			continue
-		}
-		if numaID > 63 || numaID < 0 {
-			klog.ErrorS(nil, "Invalid NUMA id range", "numaID", numaID)
+			klog.Error(err)
 			continue
 		}
+
+		numaIDToZoneIDx[numaID] = i
+
 		resources := extractResources(zone)
 		klog.V(6).InfoS("extracted NUMA resources", stringify.ResourceListToLoggable(zone.Name, resources)...)
 		nodes = append(nodes, NUMANode{NUMAID: numaID, Resources: resources})
 	}
+
+	// iterate over nodes and fill them with Costs
+	for i, node := range nodes {
+		nodes[i] = *node.WithCosts(extractCosts(zones[numaIDToZoneIDx[node.NUMAID]].Costs))
+	}
+
 	return nodes
 }
 
-func extractResources(zone topologyv1alpha2.Zone) corev1.ResourceList {
-	res := make(corev1.ResourceList)
+func getID(name string) (int, error) {
+	var numaID int
+	_, err := fmt.Sscanf(name, "node-%d", &numaID)
+	if err != nil {
+		return -1, fmt.Errorf("invalid zone format zone: %s", name)
+	}
+
+	if numaID > maxNUMAId-1 || numaID < 0 {
+		return -1, fmt.Errorf("invalid NUMA id range numaID: %d", numaID)
+	}
+
+	return numaID, nil
+}
```

Comment:
possibly offtopic here (but the stakeholder set kinda overlaps): this is something we need to better represent in the NRT API. We're depending on implicit context and on out-of-band information.

Reply:
That one confused me from the beginning :) but I guess we are stuck with it cause this is how NFD advertises `ID`.

Reply:
We should improve on NRT side, adding helpers and recommending a way to expose the zone ID: https://github.com/k8stopologyawareschedwg/noderesourcetopology-api/issues/34
For now, your current approach looks good. Let's just maybe add a couple of unit tests and it's fine.

Reply:
👍 will add unit tests on Monday.

Reply:
I have changed the implementation cause previous one was failing this testCase:
```go
		{
			description: "invalid format of name, suffix is not an integer",
			name:        "node-10a",
			expectedErr: fmt.Errorf("invalid zone format"),
		},
```
It seems that `Sscanf` was happy with matching `node-10` and just silently dropped `a`. We can also use regexp here. Let me know what you think is more suitable here.

Reply:
this is a very good catch, thanks. We need to really improve in this area (see https://github.com/k8stopologyawareschedwg/noderesourcetopology-api/issues/34 again)